mdbook-tera-backend/src/tera_renderer/renderer.rs

```diff
@@ -0,0 +1,140 @@
+use anyhow::{anyhow, Result};
+use mdbook::renderer::RenderContext;
+use serde_json::to_value;
+use std::collections::{BTreeMap, HashMap};
+use std::fs;
+use std::io::Write;
+use std::path::Path;
+use std::sync::Arc;
+use tera::Tera;
+
+/// Renderer for the tera backend.
+///
+/// This will read all the files in the `RenderContext` and render them using the `Tera` template.
+///
+/// # Example
+///
+/// ```
```

Comment:
Because this is a binary crate (there is no `src/lib.rs` file), the doctests are actually not executed. This is a problem since it gives you (and me! :smile:) a false sense of security.

Please move the doctests to unit or integration tests so that we can have confidence that the main parts of the code are tested. Right now, I see this

```
% cargo test
   Compiling mdbook-tera-backend v0.0.1 (.../mdbook-i18n-helpers/mdbook-tera-backend)
    Finished test [unoptimized + debuginfo] target(s) in 6.87s
     Running unittests src/main.rs (.../mdbook-i18n-helpers/target/debug/deps/mdbook_tera_backend-33d75d56994baca7)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
```